scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -558,8 +583,15 @@ def check_containers( self, run_state ):
 
                 for cid, info in running_containers.iteritems():
                     pod = None
-                    if cid in k8s_data:
-                        pod = k8s_data[cid]
+                    if 'k8s_info' in info:
+                        pod_name = info['k8s_info'].get( 'pod_name', 'invalid_pod' )
+                        pod_namespace = info['k8s_info'].get( 'pod_namespace', 'invalid_namespace' )
+                        pod = k8s_data.get( pod_namespace, {} ).get( pod_name, None )
+
+                        if not pod:
+                            self._logger.warning( "No pod info for container %s.  pod: '%s/%s'" % (cid[:8], pod_namespace, pod_name),
```

Comment:
You refer to cid[:8] in a bunch of places.. maybe just once in this method but several times in others.  I would create a helper function that is named whatever this is extracting out, then just invoke it on `cid` anytime you need cid[:8]... in methods where you use it a bunch of times, use a variable to hold it.  Also, add error checking in the method to handle the case where `len(cid) < 8` -- I know it probably can't, but just to be safe.